vue-frontend/src/components/Authentication.vue

```diff
@@ -108,13 +126,36 @@ export default {
       expired: state => state.status == AuthenticationStatus.EXPIRED,
       rejected: state => state.status == AuthenticationStatus.REJECTED,
       loggedOut: state => state.status == AuthenticationStatus.LOGGED_OUT,
-      message: state => state.message,
+      unauthenticated: state => state.status == AuthenticationStatus.UNAUTHENTICATED,
+      studentMessage: state => state.studentMessage,
+      educatorSignInMessage: state => state.educatorSignInMessage,
+      educatorRegisterMessage: state => state.educatorRegisterMessage,
     }),
     ...mapState('institution', {
       emailDomain: state => state.emailDomain
     }),
+    validateEmail() {
+      return this.educators.username.indexOf(this.emailDomain) !== -1 || 'Must use your'+this.emailDomain+'email.';
+    },
     validateConfirmPassword() {
       return this.educators.confirmPassword == this.educators.password || 'Must be the same as your password.';
+    },
+    studentLoading() {
+      if (this.students.username != '' && this.students.password != '') {
```

Comment:
It might be better to just add a dimension onto the authentication state which governs the credential origin. Maybe call it `state.type` and have `AuthenticationType.STUDENT` and `AuthenticationType.EDUCATOR`? I think a lot of this logic would get simplified if that were there.